File: src/java/org/javaturk/dp/ch05/facade/computer/BIOS.java
package org.javaturk.dp.ch05.facade.computer;

public class BIOS {
	private ROM rom;
	
	public BIOS() {
		rom = new ROM(2*1024);
	}
	
	public void start() {
		System.out.println("BIOS: Getting the image of OS kernel.");
		rom.getOSKernelImage();
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/builder/computer/domain/bios.ts
import {ROM} from "./rom";

export class BIOS {
  private rom: ROM;

  constructor() {
    this.rom = new ROM(2 * 1024);
  }

  start(): void {
    console.log("BIOS: Getting the image of OS kernel.");
    this.rom.getOSKernelImage();
  }
}